src/features/vote/ActiveViewDetailsModal.tsx

```diff
@@ -38,24 +43,85 @@ interface Props {
   timestamp: string;
   ancData: string;
   unix: string;
+  committedVotes: VoteEvent[] | undefined | void;
+  revealedEvents: VoteRevealed[] | undefined | void;
+  roundId: string;
 }
 
 const NULL_ANC_DATA = "0x";
+const NULL_NUM_COMMITTED_VOTES = 0;
+const NULL_PERCENTAGE = "0.00";
 
 const _ActiveViewDetailsModal: ForwardRefRenderFunction<
   HTMLElement,
   PropsWithChildren<Props>
 > = (
-  { isOpen, close, proposal, setModalState, timestamp, ancData, unix },
+  {
+    isOpen,
+    close,
+    proposal,
+    setModalState,
+    timestamp,
+    ancData,
+    unix,
+    committedVotes,
+    revealedEvents,
+    roundId,
+  },
   externalRef
 ) => {
+  const [numberCommittedVotes, setNumberCommittedVotes] = useState(
+    NULL_NUM_COMMITTED_VOTES
+  );
+  const [numberRevealedAddresses, setNumberRevealedAddresses] = useState(0);
+  const [revealPercentage, setRevealPercentage] = useState(NULL_PERCENTAGE);
   const [copySuccess, setCopySuccess] = useState(false);
   // Note: because there is dynamic content, this will rebuild the tooltip for addressing the conditional
   // elements on the page. See ReactTooltip docs.
   useEffect(() => {
     ReactTooltip.rebuild();
   });
 
+  useEffect(() => {
+    if (committedVotes && committedVotes.length && roundId && proposal) {
+      const findCommitsForProposal = committedVotes.filter(
+        (x) =>
+          x.identifier === proposal && x.roundId === roundId && x.time === unix
+      );
+      const uniqueCommitsByAddress = findCommitsForProposal.filter(
+        (v, i, a) => a.findIndex((t) => t.address === v.address) === i
```

Comment:
i might need an explanation of this logic, its a bit obscure, but it looks like you are trying to only count addresses once, and ignore duplicates? If this is the case you might want to try a [js set](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set). You can initialize the set with an array of all addresses, and then use `.size` to get the length of only uniques

Reply:
If this is removing duplicates, you can do `const unique = [...new Set(array)];` as David pointed out

Reply:
Correct. We want to see how many people have voted -- we don't really care if they have resubmitted a vote.